src/oauth/endpoints/userinfo.ts

```diff
@@ -0,0 +1,33 @@
+import { Hono } from "hono";
+import { type Variables, scopeRequired, tokenRequired } from "../middleware";
+
+const app = new Hono<{ Variables: Variables }>();
+
+app.on(["GET", "POST"], "/", tokenRequired, scopeRequired(["profile"]), (c) => {
+  const accountOwner = c.get("token").accountOwner;
+
+  if (!accountOwner) {
+    return c.json(
+      {
+        error: "This method requires an authenticated user",
```

Comment:
This aligns with the mastodon endpoint, but isn't typical of OAuth, where `error` is typically a machine value, and the descriptive value goes in `error_description`